solr/core/src/java/org/apache/solr/store/blob/metadata/ServerSideMetadata.java

```diff
@@ -214,23 +228,23 @@ public String getDirectoryHash() {
 
   /**
    * Returns <code>true</code> if the contents of the directory passed into this method is identical to the contents of
-   * the directory of the Solr core of this instance, taken at instance creation time.<p>
+   * the directory of the Solr core of this instance, taken at instance creation time. If the directory hash was not 
+   * computed at the instance creation time, then this returns <code>false</code>
```

Comment:
If directory hash was not computed I would throw IllegalStateException instead of returning false. This will help identify the programming error faster.